packages/app/src/vis-packs/nexus/containers/NxComplexImageContainer.tsx

```diff
@@ -35,17 +44,21 @@ function NxComplexImageContainer(props: VisContainerProps) {
           nxData={nxData}
           selection={getSliceSelection(dimMapping)}
           render={(nxValues) => {
-            const { signal, axisMapping, title } = nxValues;
+            const { signal, axisValues, title } = nxValues;
 
             return (
               <MappedComplexVis
                 value={signal}
                 dims={dims}
                 dimMapping={dimMapping}
-                axisMapping={axisMapping}
+                abscissaLabel={axisLabels[xDimIndex]}
+                abscissaValue={axisValues[xDimIndex]}
+                ordinateLabel={axisLabels[yDimIndex]}
+                ordinateValue={axisValues[yDimIndex]}
```

Comment:
It would make more sense to me that all the slicing/mapping happens in the `Mapped...` components. The current implementation feels conceptually wrong from the `Mapped...` component: on one hand, the axes mapping is already used to set abscissa/ordinate `labels` and `values` but on the other, we still need it to use it in the component to slice the dataset value.

So I would be in favour of computing `xDimIndex` in both components. Both serves different purposes.

Reply:
Fair enough, I'll refactor this part :+1:

Reply:
Much better indeed. If you want to have another look, please go ahead :wink: